### ISSUE TITLE
LAB04 Template not passing validation

### ISSUE THREAD

**Issue by santiagoroman** [MicrosoftLearning/AZ500-AzureSecurityTechnologies#110]
# Module: 00
#### Step: 09-10

Description of issue: All of the current participants of the AZ500 microsoft led training could not deploy the template+parameters file as they are in this repository as they do not pass validation.

The solution is to edit the template `az-500-04_azuredeploy.json` file and modify the vmSize block to

```json
"vmSize": {
    "type": "string",
    "defaultValue": "Standard_D2s_v3",
    "metadata": {
        "description": "Virtual machine size"
    }
},
```

Repro steps:

1. Follow steps in task 1 of lab 04 error will appear in steps 9-10.

**Comment by polichtm** [comment]
Hi Santiago,
just to make sure, are you referring to adding the default value to the vmSize parameter? 
This value is defined in the corresponding parameters file - have you included it in your deployment?
rgds
Marcin

**Comment by santiagoroman** [comment]
@polichtm Thanks for the quick reply! Yes, sorry I didn't clarify that. The quickest solution that another attendee of the training was able to find was to add the `defaultValue": "Standard_D2s_v3",` line to the main template file (not the parameters file).
The way it is right now in the default branch of the repository, nobody was able to just copy the template and parameters file and get it deployed through the Azure Portal as it would not pass validation for the `vmSize`.

**Comment by polichtm** [comment]
Hi Santiago,
can you clarify what you mean by "nobody was able to just copy the template and parameters file and get it deployed through the Azure Portal"?
The option to add both is available directly from the Azure portal (as per instructions).
What specifically was the validation error?
Can you provide specific repro steps?
rgds
Marcin

**Comment by santiagoroman** [comment]
Reproduction steps:

Follow steps 1 to 8 (included) just as they appear in the instructions of Task 1 (I know initially I said we were copy pasting the contents of the json files, which I was. I tried and the same thing happens if I load the file from my local disk as it's specified in the instructions).

On step 9 I noticed that I'm not able to change the VM Size parameter (if you look at the screenshot below it does appear sort of greyed out)

![image](https://user-images.githubusercontent.com/9064060/110376301-cc0ba480-8031-11eb-87c6-6dd679ebf6e6.png)

Initially I thought that this might have been due to me using Firefox and ad blocker. So I disabled all ad-blocking extensions and it still didn't allow me to edit that parameter, same thing happens using Google Chrome without any extensions enabled. I am also using a Linux based operating systems, but it should be noted that the other attendees of the training were using the official learnondemand.net platform which are Windows based and were facing the exact same issue.

Either way, even though I am unable to change this parameter I am able to click the Review + Create button so I do that considering that the default value for the vmSize as specified in the `az-500-04_azuredeploy.parameters.json` file is the one recommended in the instructions.

Once I go to the `Review + Create` step I get the following validation error:

![image](https://user-images.githubusercontent.com/9064060/110375975-64edf000-8031-11eb-8920-b99488ee2ad6.png)

**Comment by polichtm** [comment]
Hi Santiago,
here is what I'm seeing (using Firefox) - in short, on the Custom deployment blade, there is a Change size link that you can click on, which automatically displays another blade where you can select the appropriate VM size
![image](https://user-images.githubusercontent.com/10284906/110378731-ed659400-8023-11eb-8db9-0a1d789fea88.png)

![image](https://user-images.githubusercontent.com/10284906/110378779-02422780-8024-11eb-8461-a936b53236e5.png)


![image](https://user-images.githubusercontent.com/10284906/110378833-14bc6100-8024-11eb-9e5e-5dc886688f27.png)

rgds
Marcin

**Comment by santiagoroman** [comment]
Well this is going to sound crazy but I just tried again after looking at your comment and after loading the 2 `json` files I do get the default value for the VM Size instead of the greyed out version.

Normally I'd say that this may have been because Microsoft training students get a free account and because of resource prioritization in EastUS our accounts were momentarily not being allowed to deploy in that region but with the fix I suggested in the body of this issue we were able to get past the validation error so that makes me confused.

Either way, I think we can close this issue since I'm able to get the Task 1 completed just like they appear in the step by step instructions now.
